docs/DeveloperGuide.md

```diff
@@ -1,34 +1,561 @@
+first determines the main command in the user command string. If it is one of the 5 commands with a xCommandParser
+subclass, then Parser delegates the remaining work to the subclass due to the complicated logic involved. Otherwise, it
+handles the logic itself.
+
+It will create a Command object, no matter whether the user command is valid or not (if it is not, then an
+IncorrectCommand object is created). This Command object passes back to the main class Duke for execution.
+
+![UML class diagram for Parser Family Classes](https://github.com/AY2021S1-CS2113T-T09-2/tp/blob/master/docs/diagrams/ParserClassDiagram.png?raw=true)
+
+
+### Timetable Family 
+
+The Timetable Family of classes is a _cross-family_ family of classes from the Data and Command families, 
+and consists of the timetable Command and CommandParser classes, as well as TimeTableManager and TimeTable themselves. 
+Extending from the abstract TimeTableCommand class are the TimeTableAddCommand, TimeTableDeleteCommand and TimeTableViewCommand classes.
+
+![Class diagram for TimeTable Family Classes](https://github.com/AY2021S1-CS2113T-T09-2/tp/blob/master/docs/diagrams/TimeTableClassDiagram.png?raw=true)
+
+**This is a good representation of how the other data classes (Task, Module) work as well.** The Command objects
+call the methods held in the Manager classes to perform work on the stored user data.
+
+Upon the first start up of ra.VI, `TimeTableManager.initialise()` will be run. This will no longer run again as long as the user 
+does not tamper with or delete the files in the created data folder.
+
+The TimeTable is created based on the user's initial input, with an appropriate number of LessonManagers.
+The point of entry for this feature will be at TimeTableCommandParser, which will decide which of the commands 
+to return through `parseTimeTableCommand()`. If the TimeTableCommand is returned and executed, the 
+TimeTableManager will carry out the associated commands, adding, deleting or viewing the Lessons in the timetable.
+
+## Feature explanation with sequence diagrams
+
+### Main loop sequence
+When ra.VI runs, there are 3 phases to its lifecycle.
+
+1. Initialisation
+2. Main command loop
+3. Saving and exiting
+
+The main sequence diagram can be broken into three parts representing each of these phases.
+
+![Sequence diagram 1 for Main loop](https://github.com/AY2021S1-CS2113T-T09-2/tp/blob/master/docs/diagrams/MainSequenceDiagram1.png?raw=true)
+ra.VI's `start()` method initialises all the classes used by the main class, like TextUi, InputOutputManager, and more.
+TimeTableManager may need user input to be initialised, and so it is placed in a validation loop.
+A welcome message is shown at the end of the initialisation phase.
+
+![Sequence diagram 2 for Main loop](https://github.com/AY2021S1-CS2113T-T09-2/tp/blob/master/docs/diagrams/MainSequenceDiagram2.png?raw=true)
```

Comment:
I like how there are comments on the sequence diagram to explain what's going on.

Reply:
The activation bar on both ends should be closed and not left protruding out of the picture.